From 4e1deb2c49821d24cc9af55a36a8459833c8f27f Mon Sep 17 00:00:00 2001
From: Ting Liu <ting.liu@nxp.com>
Date: Thu, 18 May 2023 17:02:38 +0800
Subject: [PATCH] CMakeLists.txt: limit to wayland/*.c files

Signed-off-by: Ting Liu <ting.liu@nxp.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fadccaa..ab2940b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@ project(lv_drivers HOMEPAGE_URL https://github.com/lvgl/lv_drivers/)
 # Option to build as shared library (as opposed to static), default: OFF
 option(BUILD_SHARED_LIBS "Build shared as library (as opposed to static)" OFF)
 
-file(GLOB_RECURSE SOURCES ./*.c)
+file(GLOB_RECURSE SOURCES ./wayland/*.c)
 
 if (BUILD_SHARED_LIBS)
   add_library(lv_drivers SHARED ${SOURCES})
-- 
2.25.1

