From 1d62ec1c9a9b5f7cadf28b2b70b257fe857624df Mon Sep 17 00:00:00 2001
From: Ting Liu <ting.liu@nxp.com>
Date: Thu, 11 May 2023 16:36:58 +0800
Subject: MPU: add wm_capabilities

Signed-off-by: Ting Liu <ting.liu@nxp.com>
---
 wayland/wayland.c        | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/wayland/wayland.c b/wayland/wayland.c
index 11ed05cf3..97a043b47 100644
--- a/wayland/wayland.c
+++ b/wayland/wayland.c
@@ -220,6 +220,7 @@ struct window
 #if LV_WAYLAND_XDG_SHELL
     struct xdg_surface *xdg_surface;
     struct xdg_toplevel *xdg_toplevel;
+    uint32_t wm_capabilities;
 #endif
 
     struct buffer_allocator allocator;
@@ -1144,10 +1145,23 @@ static void xdg_toplevel_handle_configure_bounds(void *data, struct xdg_toplevel
      */
 }
 
+static void xdg_toplevel_handle_wm_capabilities(void *data, struct xdg_toplevel *xdg_toplevel,
+                                                struct wl_array *capabilities)
+{
+    uint32_t *cap;
+    struct window *window = (struct window *)data;
+
+    wl_array_for_each(cap, capabilities) {
+        window->wm_capabilities |= (1 << (*cap));
+        /* TODO: Disable appropriate graphics/capabilities as appropriate */
+    }
+}
+
 static const struct xdg_toplevel_listener xdg_toplevel_listener = {
     .configure = xdg_toplevel_handle_configure,
     .close = xdg_toplevel_handle_close,
-    .configure_bounds = xdg_toplevel_handle_configure_bounds
+    .configure_bounds = xdg_toplevel_handle_configure_bounds,
+    .wm_capabilities = xdg_toplevel_handle_wm_capabilities
 };
 
 static void xdg_wm_base_ping(void *data, struct xdg_wm_base *xdg_wm_base, uint32_t serial)
-- 
2.25.1

